### ISSUE TITLE
Connection to BaseX Query Processor fails

### ISSUE THREAD

**Issue by jblumtritt** [rhdunn/xquery-intellij-plugin#130]
Hi!
I'm experiencing some problems when trying to set up BaseX as Query Processor.
The process freezes with the message "Validating the query processor instance":

![xquery-intellij](https://user-images.githubusercontent.com/3205396/61228244-2ef64c80-a726-11e9-8e61-44c57ad1db6a.jpg)

BaseX server is running on port 1984.

I happens both with WebStrom 2019.1.3. + xquery-intellij-plugin 1.4.1 on Windows 10 and Ubuntu 18.04 (Snap Package).

Is this something you could help with?

**Comment by rhdunn** [comment]
If you close both dialogs to go back to the main IntelliJ window, is there an exception on the bottom right of the window (a red circle with an exclamation)? If there is, you can click on the icon to display the exception details.

**Comment by jblumtritt** [comment]
the exception reads:

```
java.lang.NoClassDefFoundError: org/apache/commons/io/IOUtils
	at uk.co.reecedunn.intellij.plugin.core.io.InputStreamKt.decode(InputStream.kt:23)
	at uk.co.reecedunn.intellij.plugin.core.io.InputStreamKt.decode$default(InputStream.kt:22)
	at uk.co.reecedunn.intellij.plugin.core.parser.Markdown.parse(Markdown.kt:50)
	at uk.co.reecedunn.intellij.plaugin.core.codeInspection.Inspection.getStaticDescription(Inspection.kt:28)
	at com.intellij.codeInspection.ex.InspectionToolWrapper.getStaticDescription(InspectionToolWrapper.java:157)
	at com.intellij.codeInspection.ex.InspectionToolWrapper.loadDescription(InspectionToolWrapper.java:161)
	at com.intellij.codeInspection.ex.InspectionSearchableOptionContributor.processOptions(InspectionSearchableOptionContributor.java:44)
	at com.intellij.ide.ui.search.SearchableOptionPreloader.preload(SearchableOptionPreloader.java:49)
	at com.intellij.openapi.application.Preloader.lambda$null$0(Preloader.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:586)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:86)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.application.Preloader.lambda$initComponent$1(Preloader.java:63)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:220)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:26)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.lambda$run$0(BoundedTaskExecutor.java:198)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:224)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:194)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.io.IOUtils PluginClassLoader[uk.co.reecedunn.intellij.plugin.xquery, 1.4.1] com.intellij.ide.plugins.cl.PluginClassLoader@2702451a
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:51)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 23 more
```

**Comment by rhdunn** [comment]
I've fixed the exception you posted in the current development version, but that does not appear to be related to the issue you reported. I have reproduced this with WebStorm and confirmed the fix; it does not affect IDEA.

I have also reproduced the hang you are seeing on WebStorm with the 1.4.1 version of my plugin. Testing with the current development version, I don't see the issue, so it must have been fixed at some point (I did make some changes to the BaseX support).

I'm intending on creating a new release once 2019.2 RC is released, and make it available for older IntelliJ versions as well.

I'll keep this open until you have confirmed it is fixed, e.g. after I update the plugin.

Thanks for the bug report.
